src/workerd/io/container-client.c++

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "container-client.h"
+
+#include <workerd/io/container.capnp.h>
+#include <workerd/io/docker-api.capnp.h>
+#include <workerd/jsg/jsg.h>
+
+#include <capnp/compat/json.h>
+#include <capnp/message.h>
+#include <kj/async.h>
+#include <kj/compat/http.h>
+#include <kj/debug.h>
+#include <kj/encoding.h>
+#include <kj/exception.h>
+
+namespace workerd::io {
+
+template <typename T>
+typename T::Builder decodeJsonResponse(kj::String response) {
```

Comment:
```suggestion
typename T::Builder decodeJsonResponse(kj::StringPtr response) {
```

Passing `response` as a `kj::StringPtr` should be safe here to avoid needing to move at the callsites